clang/lib/AST/Interp/EvaluationResult.cpp

```diff
@@ -122,22 +122,20 @@ static bool CheckFieldsInitialized(InterpState &S, SourceLocation Loc,
   }
 
   // Check Fields in all bases
-  for (const Record::Base &B : R->bases()) {
+  unsigned BaseIndex = 0;
+  const CXXRecordDecl *CD = dyn_cast<CXXRecordDecl>(R->getDecl());
```

Comment:
```suggestion
  const auto *CD = dyn_cast<CXXRecordDecl>(R->getDecl());
```

Reply:
This either needs to be a `cast<>` or a null check. And I think it should be a null check.

Reply:
Done.